graalvm-native-image-app/readme.md

```diff
@@ -0,0 +1,71 @@
+# graalvm-native-image-app
+
+## About
+This is an example of a native application that uses the driver and is built using
+[GraalVM native image](https://www.graalvm.org/latest/reference-manual/native-image/).
+
+## Contributor Guide
+
+This guide assumes you are using a shell capable of running [Bash](https://www.gnu.org/software/bash/) scripts.
+
+### Prepare the development environment
+
+#### Install GraalVM
+
+[GraalVM for JDK 21 Community](https://github.com/graalvm/graalvm-ce-builds/releases/tag/jdk-21.0.2) is required
+in addition to the JDK you are using for running [Gradle](https://gradle.org/) when building the driver.
+Note that GraalVM for JDK 21 Community is [available](https://sdkman.io/jdks#graalce) via [SDKMAN!](https://sdkman.io/).
+
+##### Explanation of the requirement
+
+* GraalVM Community is the only distribution of GraalVM for which it is possible to
+  specify a Gradle toolchain specification that matches only GraalVM
+  and does not match any other JDK.
+* GraalVM for Java SE 21 is required because it is the latest released version at the moment,
+  and not supporting the build for multiple, especially older versions, simplifies things.
+  Releases of JDKs for Java SE 21 having a long-term support from most vendors
+  also makes this version more attractive.  
+
+#### Configure environment variables pointing to JDKs.
+
+Assuming that the JDK you are using for running Gradle is for Java SE 17, export the following variables
+(your values may differ):
+
+```bash
+export JDK17="/Users/valentin.kovalenko/.sdkman/candidates/java/17.0.10-librca/"
+export JDK21_GRAALVM="/Users/valentin.kovalenko/.sdkman/candidates/java/21.0.2-graalce/"
```

Comment:
No, as the values with `~` are fed to `./gradlew` as is when we do `JAVA_HOME="${JDK17}"`, and it does not understand what `~` in `JAVA_HOME` is supposed to mean. `~` works in Bash because of the [tilde expansion](https://www.gnu.org/software/bash/manual/bash.html#Tilde-Expansion-1).

However, the following works provided that `realpath` is available:

> export JDK17=$(realpath ~/".sdkman/candidates/java/17.0.10-librca/")